### ISSUE TITLE
ParseDomainHint technicalprofileID does not exist and HRD not working

### ISSUE THREAD

**Issue by rodrigoffonseca** [azure-ad-b2c/samples#24]
I found a issue with **home-realm-discovery-modern** sample, **SignIn_smart_HRD.xml** file
After I edited it with my tenant and claims provider information and tried to upload it to IEF i Got the error **"ParseDomainHint is expected to be a technical profile but it does not exist"**

**Comment by rodrigoffonseca** [comment]
To provide more details about it. I found a problem while uploading the policy where it says ParseDomainHint technicalprofilereferendid was expected but was not found, so I reviewed and the code have a HRDLogic technical profile, and I modified it as below (commented line was the original code)

`<OrchestrationStep Order="2" Type="ClaimsExchange">
                    <ClaimsExchanges>
                        
                        <ClaimsExchange Id="ParseDomainHint" TechnicalProfileReferenceId="HRDLogic" />
                    </ClaimsExchanges>
                </OrchestrationStep>`

After that I got to upload it successful and test it but I still got some weird behavior.
I have to claims provider: One Azure AD Tenant (OIDC) and one ADFS (SAML), each one using different domains. When I test the HRD works fine for ADFS domain, but for AAD domain it sends me to a selection screen where I have to select AAD or ADFS provider, and I wasn't able to find what is wrong.
